File: School42/playground/-/languages/C/-/ft_strcpy.c
#include <stdio.h>

int ft_strcpy_pointers(s, t) /* copy t to s; pointer version 2 */
char *s, *t;
{
  while ((*s++ = *++t));
}

int main()
{
    char source[] = "Hello, World!";
    char destination[20];

    ft_strcpy_pointers(destination, source);

    printf("Source string: %s\n", source);
    printf("Copied string: %s\n", destination);

    return 0;
}

File: playground/python/libft_py/ft_strcpy.py
def strncpy(dest, src, n):
    dest[:n] = src[:n]
    return dest

source = "Hello, world!"
destination = bytearray(13)  # A bytearray to hold the destination

print("Before")
print("     Source:", source)
print("Destination:", destination)
# Copy at most 11 characters from source to destination
n = min(len(source), len(destination))
result = strncpy(destination, source.encode(), n)

# Convert the result back to a string for printing
result_str = result.decode()
print("\nAfter")
print("     Source:", source)
print("Destination:", result_str)
